File: DynamoDBTutorial/src/test/java/com/gkatzioura/dynamodb/supervisor/SupervisorRepositoryTest.java
package com.gkatzioura.dynamodb.supervisor;

import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
import com.amazonaws.services.dynamodbv2.model.AttributeValue;
import com.gkatzioura.dynamodb.TableCreator;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.util.Map;

/**
 * Created by gkatzioura on 7/2/16.
 */
public class SupervisorRepositoryTest {

    private AmazonDynamoDB amazonDynamoDB;
    private SupervisorRepository supervisorRepository;

    @Before
    public void setUp() {
        AmazonDynamoDB amazonDynamoDB = new AmazonDynamoDBClient();
        amazonDynamoDB.setEndpoint("http://localhost:8000");
        this.amazonDynamoDB = amazonDynamoDB;
        this.supervisorRepository = new SupervisorRepository(amazonDynamoDB);
        TableCreator tableCreator = new TableCreator(amazonDynamoDB);
        tableCreator.deleteSupervisorsTable();
        tableCreator.createSupervisor();
    }

    @Test
    public void testGetSupervisor() {

        supervisorRepository.insertSupervisor("John Doe","Sun","Athens");

        Map<String,AttributeValue> item = supervisorRepository.getSupervisor("Sun","Athens");

        Assert.assertEquals("John Doe",item.get("name").getS());
    }

}

File: DynamoDBTutorialNode/test/supervisor/SupervisorRepositoryTest.js
/**
 * http://usejsdoc.org/
 */

var SupervisorRepository = require('../../supervisor/SupervisorRepository'),
	TableCreator = require('../../TableCreator'),
	assert = require('assert');

describe('Supervisor Repository Test',function(){
    
	before(function(done) {
		TableCreator.deleteSupervisors(function(err,result) {
			
			if(err) {
				test.done(err);
				return;
			}
			
			
			TableCreator.createSupervisors(function(err,result) {
				
				if(err) {
					test.done(err);
					return;
				}
				
				SupervisorRepository.insertSupervisor("john@doe.com", "Sun", "Athens", function(err,result) {
					
					if(err) {
						done(err);
					} else {
						done();
					}
				});

			});			
		});
	});
	
	it('Query supervisor',function(done) {
		SupervisorRepository.getSupervisor("Sun", "Athens",function(err,result) {
			if(err) {
				done(err);
			} else {
				console.log(result);
				done();
			}
		});
	});
	
})

